comp/core/agenttelemetry/impl/agenttelemetry.go

```diff
@@ -128,9 +129,13 @@ func createAtel(
 
 // NewComponent creates a new agent telemetry component.
 func NewComponent(req Requires) agenttelemetry.Component {
+	if !pkgconfigsetup.IsAgentTelemetryEnabled(req.Config) {
+		return &atel{enabled: false}
+	}
```

Comment:
This check makes lines 113-116 redundant (and the log message would be never printed). Would it make sense to refactor this to check once? Perhaps by moving sender and runner creation into `createAtel`.

Reply:
You commented on the area I did not like myself. I contemplating a few ways to address, including one you have suggested, and did not like any since createAtel is used by both code and unit tests. However, passing nil as default parameters into the helper routing as you have suggested, would be IMHO a decently enough change. This is what I have done in the latest commit. Thank you.